[Merged by Bors] - fix(tactic/squeeze_simp): make `squeeze_simp [←...]` work

Vierkantor opened the pull request:
`squeeze_simp` parses the argument list using a function in core Lean, and when support for backwards arguments was added to `simp`, it used a new function to parse the additional structure. This PR fixes the TODO left in the code to switch `squeeze_simp` to the new function by deleting the code that needed it - it wasn't used anyway!

To add a test for the fix, I moved the single existing `squeeze_simp` test from the deprecated file `examples.lean` to a new file.

Vierkantor committed:
fix(tactic/squeeze_simp): make `squeeze_simp [←...]` work

`squeeze_simp` parses the argument list using a function in core Lean, and when
support for backwards arguments was added to `simp`, it used a new function to
parse the additional structure. This PR fixes the TODO left in the code to
switch `squeeze_simp` to the new function too, allowing use of backwards
arguments.

gebner commented on src/tactic/squeeze.lean:
```diff
@@ -30,10 +30,9 @@ namespace tactic
-    match h.get_app_fn h with
+  (hs, _, _) ← decode_simp_arg_list_with_symm hs,
+  pure $ hs.foldr (λ (h : pexpr × bool) (s : name_set),
+    match h.1.get_app_fn h.1 with
```
Can you explain why this match works?  At first glance it looks like `h.1.get_app_fn h.1` elaborates to `⇑h.1.get_app_fn h.1` and hence never matches `expr.const _ _` because it is an `expr.app`.

Vierkantor replied:
I believe you are correct that the match doesn't work, but I'm unsure what the exact intention of the function is (the match was like this when this code was introduced in PR #1047). It should remove duplicates when a `@[simp]` lemma is explicitly added to the arguments, right? But I haven't been able to construct a test case for `squeeze_simp` where I saw a difference between `h.1.get_app h.1` and with `h.1.get_app` - both versions remove duplicate/unneeded lemmas.

gebner replied:
Can we just erase the `erase_simp_args` function then?

Vierkantor replied:
Looks like #2218 changed `squeeze_simp` so that any duplicate/unneeded lemmas will always get deleted by the call to `filter_simp_set` after that. My conclusion: erase `erase_simp_args`.

Vierkantor committed:
Erase `erase_simp_args`

This function was used to deduplicate/filter the result of `squeeze_simp`,
but it appears that `filter_simp_set` already takes care of this since #2218.

Co-Authored-By: Gabriel Ebner <gebner@gebner.org>
